### PR TITLE
Factor in total current consumption in battery remaining estimation

### LINKED ISSUE #16932: BATx_CAPACITY not considered in remaining percentage indication
When setting the BAT1_CAPACITY the expected behavior is:
1. Calculate remaining percentage based on voltage
2. Calculate remaining percentage based on battery capacity
3. Indicate whichever provides the lower percentage

Currently it seems only the voltage is taken into account

image

### PR DESCRIPTION
**Pull request by sanderux**
Closes #16932

### CODE REVIEW

File: src/lib/battery/battery.cpp [hunk 1]
```diff
@@ -226,6 +226,12 @@ void Battery::estimateRemaining(const float voltage_v, const float current_a, co
 			// directly apply current capacity slope calculated using current
 			_remaining -= _discharged_mah_loop / _params.capacity;
 			_remaining = math::max(_remaining, 0.f);
+
+			// calculation should never exceed percentage based on raw current consumption
+			_remaining_current = 1 - _discharged_mah / _params.capacity;
```
Comment (dagar) [inline src/lib/battery/battery.cpp:231]: Put all of this within a check that `_params.capacity` (BATn_CAPACITY) is > 0?

Comment (sanderux) [reply src/lib/battery/battery.cpp:231]: Its already within that if statement

Comment (sanderux) [reply src/lib/battery/battery.cpp:231]: https://github.com/PX4/PX4-Autopilot/blob/73ab87e4f127ada15f99ef4ef90c85fc3ec01ad1/src/lib/battery/battery.cpp#L216

**Review by RomanBapst — APPROVED** [review]

File: src/lib/battery/battery.cpp [hunk 2]
```diff
@@ -210,41 +210,44 @@ void Battery::estimateRemaining(const float voltage_v, const float current_a, co
...
-			_remaining = (1 - weight_v) * _remaining + weight_v * _remaining_voltage;
+			const float weight_v = 3e-4f * (1 - _state_of_charge_volt_based);
+			_state_of_charge = (1 - weight_v) * _state_of_charge + weight_v * _state_of_charge_volt_based;
 			// directly apply current capacity slope calculated using current
-			_remaining -= _discharged_mah_loop / _params.capacity;
-			_remaining = math::max(_remaining, 0.f);
+			_state_of_charge -= _discharged_mah_loop / _params.capacity;
+			_state_of_charge = math::max(_state_of_charge, 0.f);
+
+			const float state_of_charge_current_based = math::max(1 - _discharged_mah / _params.capacity, 0.f);
```
Comment (dagar) [inline src/lib/battery/battery.cpp:230]:
```suggestion
			const float state_of_charge_current_based = math::max(1.f - _discharged_mah / _params.capacity, 0.f);
```

**Review by dagar — APPROVED** [review]

**Pull request merged**